CMP1903M Assessment 1 Base Code/Analyse.cs

```diff
@@ -6,31 +6,149 @@
+        // Such as 'public List<int> analyseText(string input)'
+        // These methods that can be invoked outside the class are the only things which can access the private variables
         public List<int> analyseText(string input)
         {
-            //List of integers to hold the first five measurements:
-            //1. Number of sentences
-            //2. Number of vowels
-            //3. Number of consonants
-            //4. Number of upper case letters
-            //5. Number of lower case letters
-            List<int> values = new List<int>();
-            //Initialise all the values in the list to '0'
+            List<int> Values = new List<int>();
             for(int i = 0; i<5; i++)
             {
-                values.Add(0);
+                Values.Add(0);
+            }
+
+            if(input == null)
+            {
+                throw new ArgumentException("Invalid Input.");
+            }
+            // Finding the number of sentences
+            char[] delimiter = {'.', '!', '?'};
+            string[] sentences = input.Split(delimiter);
+            foreach (var item in sentences)
+            {
+                if (item.Trim() != "")
+                {
+                    numberOfSentences++;
+                }
+            }
+            Values[0] = numberOfSentences;
+
+            // Finding the vowels and consonants
+            foreach (var item in input)
+            {
+                if (vowelArray.Contains(item))
+                {
+                    numberOfVowels++;
+                }
+                else if (consonantArray.Contains(item))
+                {
+                    numberOfConsonants++;
+                }
             }
+            Values[1] = numberOfVowels;
+            Values[2] = numberOfConsonants;
 
+            // Finding the Upper-case and Lower-case letters;
+            string lowerArray = upperArray.ToLower();
+            foreach (var item in input)
+            {
+                if (upperArray.Contains(item))
+                {
+                    numberOfUpperCase++;
+                }
+                else if (lowerArray.Contains(item))
+                {
+                    numberOfLowerCase++;
+                }
+            }
+            Values[3] = numberOfUpperCase;
+            Values[4] = numberOfLowerCase;
 
-            return values;
+            return Values;
         }
+        // Method to find all words longer than 7 letters 
+        public void findLongWords(string input)
+        {   // Creation of list of long words
+            List<string> longWords = new List<string>();
+            // Temporary string to hold words
+            string words = "";
+            foreach (char item in input.ToCharArray())
+            {  // If a real character is found then adds it to the temp string
+               if (char.IsLetterOrDigit(item))
+               {
+                   words += item.ToString();
```

Comment:
Instead of an `if-else`, you could `continue` after this line. 

This means that the else block is indented less, making it more readable. And is good practice for larger codebases. 

If you Google the concept "Happy path", you can find out more.

https://en.wikipedia.org/wiki/Happy_path

Reply:
So instead of
```c#
if (...) {
    // code
} else {
   // code
}
```

you can do
```c#
if (...) {
    // code
    continue;
}
// else code
```

Reply:
Ohh wow yeah!  I just changed it in my code and it looks much nicer, thank you!